src/main/java/it/pagopa/transactions/commands/handlers/TransactionRequestAuthorizationHandler.java

```diff
@@ -175,13 +183,26 @@ public Mono<RequestAuthorizationResponseDto> handle(TransactionRequestAuthorizat
                                             )
                                     );
 
-                                    return transactionEventStoreRepository.save(authorizationEvent)
-                                            .thenReturn(tuple3)
-                                            .map(
-                                                    auth -> new RequestAuthorizationResponseDto()
-                                                            .authorizationUrl(tuple3.getT2())
-                                                            .authorizationRequestId(tuple3.getT1())
+                                    Mono<Void> updateSession = Mono.just(command.getData().authDetails())
```

Comment:
Can you unify these two blocks? 🙂
`return Mono.just(command.getData().authDetails())
                                            .filter(CardsAuthRequestDetailsDto.class::isInstance)
                                            .cast(CardsAuthRequestDetailsDto.class)
                                            .flatMap(
                                                    authRequestDetails -> paymentMethodsClient.updateSession(
                                                            command.getData().paymentInstrumentId(),
                                                            authRequestDetails.getSessionId(),
                                                            command.getData().transaction().getTransactionId().value()
                                                    )
                                            ).then(transactionEventStoreRepository.save(authorizationEvent)
                                                    .thenReturn(tuple3)
                                                    .map(
                                                            auth -> new RequestAuthorizationResponseDto()
                                                                    .authorizationUrl(tuple3.getT2())
                                                                    .authorizationRequestId(tuple3.getT1())
                                                    ));`

Reply:
I don't know, I don't feel like it would improve readability